packages/cli-platform-android/src/commands/runAndroid/listAndroidUsers.ts

```diff
@@ -0,0 +1,61 @@
+import {logger} from '@react-native-community/cli-tools';
+import {spawnSync} from 'child_process';
+import prompts from 'prompts';
+
+type User = {
+  id: string;
+  name: string;
+};
+
+export function checkUsers(device: string, adbPath: string) {
+  try {
+    const adbArgs = ['-s', device, 'shell', 'pm', 'list', 'users'];
+
+    logger.debug(`Checking users on "${device}"...`);
+    const {stdout} = spawnSync(adbPath, adbArgs, {encoding: 'utf-8'});
+    const regex = new RegExp(/UserInfo{([0-9]*):([^:]*):[\w]*}/, 'g');
```

Comment:
is there a way to list users in a json or other structured format?

Reply:
If not, and we need to keep the regex, please include a unit test with sample output from adb

Reply:
For the record I did not find any way to fetch users in a structured way.

Here is the output on my environment:

```
❯ adb -s ... shell pm list users
Users:
        UserInfo{0:Homersimpsons:c13} running
        UserInfo{10:Guest:404}
```

So I guess the best RegEx could be (https://regex101.com/r/8VjP87/1):
```
/^\s*UserInfo\{(?<userId>\d+):(?<userName>.*):(?<userFlags>[0-9a-f]*)}/
```

And in fact a test should be added.

For reference here is the link where this string is produced in Android's source (It might be interesting to include it, the format is unlikely to change given the coment):
https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/core/java/android/content/pm/UserInfo.java#461

Also the `pm` command is defined here (I could not found where the `cmd` comes from):
https://android.googlesource.com/platform/frameworks/base/+/refs/heads/master/cmds/pm/pm.sh

Reply:
Done :)